observatory-dags/observatory/dags/workflows/elastic_import_workflow.py

```diff
@@ -24,37 +24,81 @@
 from concurrent.futures import ProcessPoolExecutor
 from concurrent.futures import ThreadPoolExecutor, as_completed
 from multiprocessing import cpu_count
-from typing import List
-from typing import Optional
+from typing import List, Optional, Callable
 
 import elasticsearch.exceptions
 import google.cloud.bigquery as bigquery
 import pendulum
 from airflow.exceptions import AirflowException
 from airflow.hooks.base_hook import BaseHook
-from airflow.models import TaskInstance, Variable
+from airflow.models import TaskInstance
 from airflow.operators.sensors import ExternalTaskSensor
 from natsort import natsorted
 from pendulum import Pendulum, Date
 
 from observatory.platform.elastic.elastic import (
     Elastic,
     make_sharded_index,
-    make_elastic_uri,
-    elastic_mappings_path,
 )
-from observatory.platform.elastic.kibana import Kibana, ObjectType
+from observatory.platform.elastic.elastic import make_elastic_mappings_path
+from observatory.platform.elastic.kibana import Kibana, ObjectType, TimeField
 from observatory.platform.telescopes.snapshot_telescope import SnapshotRelease
 from observatory.platform.telescopes.telescope import Telescope
-from observatory.platform.utils.airflow_utils import AirflowVariable as Variable, AirflowVars, AirflowConns
-from observatory.platform.utils.file_utils import yield_jsonl, yield_csv
+from observatory.platform.utils.airflow_utils import AirflowVars, AirflowConns
+from observatory.platform.utils.file_utils import yield_jsonl, yield_csv, load_file, write_to_file
 from observatory.platform.utils.gc_utils import (
     bigquery_sharded_table_id,
     select_table_shard_dates,
     download_blobs_from_cloud_storage,
 )
 from observatory.platform.utils.jinja2_utils import render_template
 
+ES_INDEX_STATE_TOPIC_NAME = "es_index_state"
+CSV_TYPES = ["csv", "csv.gz"]
+JSONL_TYPES = ["jsonl", "jsonl.gz"]
+
+
+def load_elastic_mappings_simple(path: str, table_prefix: str):
+    return json.loads(load_file(os.path.join(path, f"{table_prefix.replace('_', '-')}-mappings.json")))
+
+
+def load_elastic_mappings_ao(path: str, table_prefix: str):
+    if not table_prefix.startswith("ao"):
+        raise ValueError("Table must begin with 'ao'")
+    elif table_prefix.startswith("ao_doi"):
```

Comment:
what about having an ao_simple prefix for those that want to use load_elastic_mappings_simple
otherwise you will need to keep adding if branches for new workflows